tests/integration/telemetry/stackdriver/stackdriver_filter_audit_test.go

```diff
@@ -0,0 +1,161 @@
+	"time"
+
+	"golang.org/x/sync/errgroup"
+
+	"istio.io/istio/pkg/test/framework"
+	"istio.io/istio/pkg/test/framework/components/echo"
+	"istio.io/istio/pkg/test/framework/components/stackdriver"
+	"istio.io/istio/pkg/test/scopes"
+	"istio.io/istio/pkg/test/util/file"
+	"istio.io/istio/pkg/test/util/retry"
+	"istio.io/istio/pkg/test/util/tmpl"
+	"istio.io/istio/tests/integration/telemetry"
+)
+
+const (
+	serverAuditAllLogEntry  = "testdata/security_authz_audit/server_audit_all_log.json.tmpl"
+	serverAuditFooLogEntry  = "testdata/security_authz_audit/server_audit_foo_log.json.tmpl"
+	serverAuditBarLogEntry  = "testdata/security_authz_audit/server_audit_bar_log.json.tmpl"
+	serverAuditNoneLogEntry = "testdata/security_authz_audit/server_audit_none_log.json.tmpl"
+	auditPolicyForLogEntry  = "testdata/security_authz_audit/v1beta1-audit-authorization-policy.yaml.tmpl"
+)
+
+// TestStackdriverAuditLogging testing Authz Policy can config stackdriver with audit policy
+func TestStackdriverHTTPAuditLogging(t *testing.T) {
+	framework.NewTest(t).
+		Features("observability.telemetry.stackdriver").
+		Run(func(ctx framework.TestContext) {
+			g, _ := errgroup.WithContext(context.Background())
+
+			ns := getEchoNamespaceInstance().Name()
+			args := map[string]string{
+				"Namespace": ns,
+			}
+			policies := tmpl.EvaluateAllOrFail(t, args, file.AsStringOrFail(t, auditPolicyForLogEntry))
+			ctx.Config().ApplyYAMLOrFail(t, ns, policies...)
+			defer ctx.Config().DeleteYAMLOrFail(t, ns, policies...)
+			t.Logf("Audit policy deployed to namespace %v", ns)
+
+			t.Logf("Traffic sent to namespace %v", ns)
+			for _, cltInstance := range clt {
+				scopes.Framework.Infof("Validating Audit policy and Telemetry for Cluster %v", cltInstance.Config().Cluster.Name())
+				g.Go(func() error {
+					err := retry.UntilSuccess(func() error {
+						if err := sendTrafficForAudit(t, cltInstance); err != nil {
+							return err
+						}
+
+						clName := cltInstance.Config().Cluster.Name()
+						t.Logf("Collect Audit Log for cluster %v", clName)
+
+						var errs []string
+
+						errAuditFoo := validateLogs(t, serverAuditFooLogEntry, clName, stackdriver.ServerAuditLog)
+						if errAuditFoo == nil {
+							t.Logf("Foo Audit Log validated for cluster %v", clName)
+						} else {
+							errs = append(errs, errAuditFoo.Error())
+						}
+
+						errAuditBar := validateLogs(t, serverAuditBarLogEntry, clName, stackdriver.ServerAuditLog)
+						if errAuditBar == nil {
+							t.Logf("Bar Audit Log validated for cluster %v", clName)
+						} else {
+							errs = append(errs, errAuditBar.Error())
+						}
+
+						errAuditAll := validateLogs(t, serverAuditAllLogEntry, clName, stackdriver.ServerAuditLog)
+						if errAuditAll == nil {
+							t.Logf("All Audit Log validated for cluster %v", clName)
+						} else {
+							errs = append(errs, errAuditAll.Error())
+						}
+
+						errAuditNone := validateLogs(t, serverAuditNoneLogEntry, clName, stackdriver.ServerAuditLog)
```

Comment:
Please update the none audit log verification to verify there is no audit log entry with `/audit-none` as log path, and get rid of the `server_audit_none_log.json.tmpl` file. It could simply be:
```go
entries, err := sdInst.ListLogEntries(stackdriver.ServerAuditLog)
if err != nil {
  errs = append(errs, err)
} else {
  for _, l := range entries {
    if l.httpRequest != nil && strings.HasSuffix(l.httpReuqest.requestUrl, "audit-none") {
      errs = append(errs, errors.New("unwanted audit log entry `/audit-none` recieved."))
    }
  }
}
```